artemis-commons/src/main/java/org/apache/activemq/artemis/utils/CompositeAddress.java

```diff
@@ -41,12 +44,25 @@ public static SimpleString toFullyQualified(SimpleString address, SimpleString q
       return result;
    }
 
+   /**
+    * Checks whether a given string complies the fully qualified queue name (FQQN) standard or not.
+    *
+    * @param address The string to check for FQQN standard compliance.
+    * @return boolean
+    */
    public static boolean isFullyQualified(String address) {
-      return address == null ? false : address.contains(SEPARATOR);
+      List<String> split = Splitter.onPattern(SEPARATOR).splitToList(address);
```

Comment:
This method is used a lot throughout the code-base including during message routing (which is important to performance). Therefore, the less garbage it produces the better. Here you're creating a `List` of `String` objects. I think this could be implemented more simply with something like `String.indexOf` instead. What do you think?

Reply:
And now we have a JMH module to place a nice benchmark to prove that's improved in term of garbage produced too :)

Reply:
Here's a working example:
```java
   public static boolean isFullyQualified(String address) {
      if (address == null) {
         return false;
      }
      int index = address.indexOf(SEPARATOR);
      if (index == -1 || index == 0 || index == address.length() - 2) {
         return false;
      }

      return true;
   }
```

Reply:
Hey Justin, thanks a lot for the suggestion, I had also encountered a similar piece of code in the same class which was using indexOf and was thinking of changing into it.

I re-ran the tests and the performance improvement is really noticeable so thank you again!

I am now following up on the test failures albeit from what I can see the one I was previously following up, related to the AMQP protocol (ProtonServerSendContext.java class), seems to have been fixed by these changes!